wvdecrypter/cdm/media/cdm/cdm_adapter.cc

```diff
@@ -507,7 +501,21 @@ cdm::Buffer* CdmAdapter::Allocate(uint32_t capacity)
 void CdmAdapter::SetTimer(int64_t delay_ms, void* context)
 {
   //LICENSERENEWAL
-  std::thread(timerfunc, shared_from_this(), delay_ms, context).detach();
+  // Clean up async tasks which have finished
+  {
+    std::lock_guard<std::mutex> lock(m_closeSessionMutex);
+    for (auto i = 0; i < m_futures.size(); i++) {
+      if (!m_futures[i].valid() || m_futures[i].wait_for(std::chrono::milliseconds(0)) ==
+        std::future_status::ready)
+      {
+        m_futures.erase(m_futures.begin() + i);
+        i--;
+      }
+    }
```

Comment:
why not use iterator?
```suggestion
    for (auto& itFuture = m_futures.begin(); itFuture != m_futures.end();)
    {
      if (!itFuture->valid() ||
          itFuture->wait_for(std::chrono::milliseconds(0)) == std::future_status::ready)
      {
        itFuture = m_futures.erase(itFuture);
      }
      else
        itFuture++;
    }
```